File: CS/DisplayDetailInPopupWindow/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DisplayDetailInPopupWindow.Models;

namespace DisplayDetailInPopupWindow.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }
        [HttpPost]
        public PartialViewResult MasterAction() {
            return PartialView("MasterViewPartial", CustomerRepository.GetCustomers());
        }
        public ActionResult DetailAction(string _customerID) {
            ViewData["customerID"] = _customerID;
            return View("DetailView", ViewData["customerID"]);
        }
        [HttpPost]
        public PartialViewResult DetailPartialAction(string _customerID) {
            ViewData["customerID"] = _customerID;
            return PartialView("DetailViewPartial", OrderRepository.GetOrders(_customerID));
        }
    }
}

File: VB/DisplayDetailInPopupWindow/Controllers/HomeController.vb
Imports System.Web.Mvc
Imports DisplayDetailInPopupWindow.Models

Namespace DisplayDetailInPopupWindow.Controllers

    Public Class HomeController
        Inherits Controller

        Public Function Index() As ActionResult
            Return View()
        End Function

        <HttpPost>
        Public Function MasterAction() As PartialViewResult
            Return PartialView("MasterViewPartial", GetCustomers())
        End Function

        Public Function DetailAction(ByVal _customerID As String) As ActionResult
            ViewData("customerID") = _customerID
            Return View("DetailView", ViewData("customerID"))
        End Function

        <HttpPost>
        Public Function DetailPartialAction(ByVal _customerID As String) As PartialViewResult
            ViewData("customerID") = _customerID
            Return PartialView("DetailViewPartial", GetOrders(_customerID))
        End Function
    End Class
End Namespace
